### DISCUSSION TITLE
Showing line number for warnings and errors

### DISCUSSION THREAD

**Discussion by Sonico98** [nvim-lualine/lualine.nvim#721] [Q&A] [answered]
When using ALE or coc, is it possible to show the line number of the first warning and/or error encountered in the file?
I have defined a section on my config that is pretty much the same as the one shown on the README
```
lualine_b = {
	'branch',
	'diff',
	{
		'diagnostics',
		sources = { 'coc', 'ale', 'nvim_lsp' },
		sections = { 'error', 'warn', 'info', 'hint' },
		diagnostics_color = {
			error = 'DiagnosticError',
			warn  = 'DiagnosticWarn',
			info  = 'DiagnosticInfo',
			hint  = 'DiagnosticHint',
		},
		symbols = {error = 'E', warn = 'W', info = 'I', hint = 'H'},
		colored = true,
		update_in_insert = false,
		always_visible = false,
	}
  }
```
and while this does show how many errors/warnings there are, it does not display the line numbers indicating where to find them. Is there something I could add to the config for that to show up?

**Comment by shadmansaleh** [comment, marked as answer]
I think both ale & coc has their api exposing those info . You can grab that info through those and show them in lualine with custom components (like a function component)
